src/dateTime/regexParse.ts

```diff
@@ -0,0 +1,368 @@
+const isoExtendedZone = `(?:${offsetRegex.source}?(?:\\[(${ianaRegex.source})\\])?)?`;
+// hh:mm:ss.sss
+const isoTimeBaseRegex = /(\d\d)(?::?(\d\d)(?::?(\d\d)(?:[.,](\d{1,30}))?)?)?/;
+const isoTimeRegex = RegExp(`${isoTimeBaseRegex.source}${isoExtendedZone}`);
+const isoTimeOnly = RegExp(`^T?${isoTimeBaseRegex.source}$`);
+// Thh:mm:ss.sss±00:00
+const isoTimeExtensionRegex = RegExp(`(?:T${isoTimeRegex.source})?`);
+// YYYY-MM-DD
+const isoYmdRegex = /([+-]\d{6}|\d{4})(?:-?(\d\d)(?:-?(\d\d))?)?/;
+// YYYY-Www-D
+const isoWeekRegex = /(\d{4})-?W(\d\d)(?:-?(\d))?/;
+// YYYY-DDD
+const isoOrdinalRegex = /(\d{4})-?(\d{3})/;
+
+const isoYmdWithTimeExtensionRegex = new RegExp(
+    `^${isoYmdRegex.source}${isoTimeExtensionRegex.source}$`,
+);
+const isoWeekWithTimeExtensionRegex = new RegExp(
+    `^${isoWeekRegex.source}${isoTimeExtensionRegex.source}$`,
+);
+const isoOrdinalWithTimeExtensionRegex = new RegExp(
+    `^${isoOrdinalRegex.source}${isoTimeExtensionRegex.source}$`,
+);
+const isoTimeFullRegex = new RegExp(`^${isoTimeRegex.source}$`);
+
+// https://datatracker.ietf.org/doc/html/rfc2822#section-4.3
+const obsOffsets = {
+    GMT: 0,
+    UT: 0,
+    EDT: -4 * 60,
+    EST: -5 * 60,
+    CDT: -5 * 60,
+    CST: -6 * 60,
+    MDT: -6 * 60,
+    MST: -7 * 60,
+    PDT: -7 * 60,
+    PST: -8 * 60,
+};
+
+// RFC 2822/5322 https://datatracker.ietf.org/doc/html/rfc2822
+// Fri, 19 Nov 82 16:14:55 GMT
+const rfc2822 =
+    /^(?:(Mon|Tue|Wed|Thu|Fri|Sat|Sun),\s)?(\d{1,2})\s(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\s(\d{2,4})\s(\d\d):(\d\d)(?::(\d\d))?\s(?:(UT|GMT|[ECMP][SD]T)|([Zz])|(?:([+-]\d\d)(\d\d)))$/;
+
+function preprocessRFC2822(s: string) {
+    // Remove comments and folding whitespace and replace multiple-spaces with a single space
+    return s
+        .replace(/\([^()]*\)|[\n\t]/g, ' ')
+        .replace(/(\s\s+)/g, ' ')
+        .trim();
+}
+
+function extractRfc2822(
+    match: RegExpExecArray,
+): [dateObj: ExtractedDateObject, timezoneOffset: number] {
+    const [
+        ,
+        weekday,
+        day,
+        month,
+        year,
+        hour,
+        minute,
+        second,
+        obsOffset,
+        zeroOffset,
+        offsetHours,
+        offsetMinutes,
+    ] = match;
+
+    const result = stringsToDateObject(weekday, year, month, day, hour, minute, second);
+
+    let offset: number;
+    if (obsOffset) {
+        offset = obsOffsets[obsOffset as keyof typeof obsOffsets];
+    } else if (zeroOffset) {
+        offset = 0;
+    } else {
+        const hours = parseInt(offsetHours, 10);
+        const sign = hours < 0 || Object.is(hours, -0) ? -1 : 1;
+        const minutes = parseInt(offsetMinutes, 10) || 0;
+        offset = (hours || 0) * 60 + sign * minutes;
+    }
+    return [result, offset];
+}
+
+function extractISOYmd(match: RegExpExecArray, cursor: number): PartialResult {
+    const item = {
+        year: parseInteger(match[cursor]),
+        month: parseInteger(match[cursor + 1], 1) - 1,
```

Comment:
Please, add a comment abour  `- 1`.

 I think that starting the month from `0` is lit confusing.